packages/libs/error-stack/src/hook.rs

```diff
@@ -26,6 +28,168 @@ impl fmt::Display for HookAlreadySet {
+    /// #
+    /// #     let value = backtrace.replace_all(&value, "Backtrace No. $1\n  [redacted]");
+    /// #     let value = backtrace_info.replace_all(value.as_ref(), "backtrace with [n] frames ($2)");
+    /// #
+    /// #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
+    /// # }
+    /// #
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/snapshots/hook__fallback.snap")].assert_eq(&render(format!("{report:?}")));
+    /// #
+    /// # stringify!(
+    /// println!("{report:?}");
+    /// # );
+    /// ```
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/snapshots/hook__fallback.snap"))]
+    /// </pre>
+    ///
+    /// ```
+    /// use std::io::{Error, ErrorKind};
+    ///
+    /// use error_stack::{fmt, report, Report};
+    /// use error_stack::fmt::{Call, Emit};
+    ///
+    /// struct Suggestion(&'static str);
+    ///
+    /// Report::install_debug_hook_fallback(|val, ctx| {
+    ///     // first run all builtin hooks to make sure that we print backtrace and spantrace
+    ///     match fmt::builtin(val, ctx) {
+    ///         Call::Miss(_) => Call::Find(Emit::next("unknown")),
+    ///         Call::Find(emit) => Call::Find(emit),
+    ///     }
+    /// });
+    ///
+    /// let report =
+    ///     report!(Error::from(ErrorKind::InvalidInput)).attach(Suggestion("O no, try again"));
+    ///
+    /// # owo_colors::set_override(true);
+    /// # fn render(value: String) -> String {
+    /// #     let backtrace = regex::Regex::new(r"Backtrace No\. (\d+)\n(?:  .*\n)*  .*").unwrap();
+    /// #     let backtrace_info = regex::Regex::new(r"backtrace with (\d+) frames \((\d+)\)").unwrap();
+    /// #
+    /// #     let value = backtrace.replace_all(&value, "Backtrace No. $1\n  [redacted]");
+    /// #     let value = backtrace_info.replace_all(value.as_ref(), "backtrace with [n] frames ($2)");
+    /// #
+    /// #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
+    /// # }
+    /// #
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/snapshots/hook__fallback_builtin.snap")].assert_eq(&render(format!("{report:?}")));
+    /// #
+    /// # stringify!(
+    /// println!("{report:?}");
+    /// # );
+    /// ```
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/snapshots/hook__fallback_builtin.snap"))]
+    /// </pre>
+    #[cfg(feature = "std")]
+    pub fn install_debug_hook_fallback(
+        hook: impl for<'a> Fn(&Frame, HookContext<'a, Frame>) -> Call<'a, Frame> + Send + Sync + 'static,
+    ) {
+        let mut lock = FMT_HOOK.write().expect("should not be poisoned");
+        lock.fallback(hook);
+    }
+
+    /// Returns the hook that was previously set by [`install_debug_hook`]
+    ///
+    /// [`install_debug_hook`]: Self::install_debug_hook
+    #[cfg(feature = "std")]
+    pub(crate) fn with_format_hook<T>(closure: impl FnOnce(&Hooks) -> T) -> T {
```

Comment:
The name is misleading, it feels like it would create `Self` rather than returning `T`. What about `get_debug_format_hook`?